### DISCUSSION TITLE
Yarn not installing properly

### DISCUSSION THREAD

**Discussion by MED-1996** [smartcontractkit/full-blockchain-solidity-course-py#376] [Q&A] [answered]
When I run the terminal command to install yarn, yarn installs!

`mitchelldaddario@mitchells-air web3_py_simple_storage % npm install --gloabl yarn                               `

However when I check to see the version of yarn, I get the following error message...

`mitchelldaddario@mitchells-air web3_py_simple_storage % yarn --version                                          
`
`zsh: command not found: yarn`

I've been trying to troubleshoot this but can't figure it out. I did find that my home directory, on my Mac book, has a node_modules folder & I see yarn was installed there. But I also found another node_modules folder in /usr/local/lib/node_modules. In this folder, which I believe is the one that is being used by my programs, yarn was not installed. I dropped the yarn folder from my home directory into this folder but that still didn't fix my issue.

What am I doing wrong?

**Comment by cromewar** [comment, marked as answer]
Hello @MED-1996 this is because you need to add npm packages to your path, otherwise the terminal would no recognize packages like yarn, take [this](https://stackoverflow.com/questions/14803978/npm-global-path-prefix) thread as point of reference.

**Reply by MED-1996** [reply]
I feel so bad for asking all these questions...I now have 3 more questions....

1.) What is my "path"?
2.) How do I find my "path"?
3.) How do I add npm packages to my "path"?

I always hear ppl talk about "path" on forums but I don't know what the definition is.

**Reply by cromewar** [reply]
The path is a common way to refer to environment variables, they are something you assign which can be accessed globally on your computer, them could be programs, commands or simple variables like a private key for your metamask account.

The environment variables are accessed differently, depending on the operating system.. for Unix based systems like Mac or Linux you can export your variables just using the `export` key and the variable on your terminal, however that variable would last just meanwhile the terminal is open.

To add programs or packages permanently like yarn for example your best way to go is to create 'symlinks', meaning you would use a command and the command would point to the package. Please give a check the link I posted above it seems another Mac user has the same problem with them terminal not recognizing npm packages... Give it s try.

Note: ask all the questions you want, don't be afraid to ask.

**Reply by cromewar** [reply]
O yes, I almost forgot you can also add env variables editing the ~/.bashrc or ~/.zshrc file ok your home directory, however I would not recommend mess with this u lees you know what you are doing.

**Reply by MED-1996** [reply]
I actually figured it out! I was writing `--gloabl` instead of `--global` when I was installing yarn :(((

Thank you so much for the help. You're helping me out a lot!! I appreciate it :)

**Reply by cromewar** [reply]
haha this things happens a lot, don't worry. Good luck with the rest of the course ;)
